### PR TITLE
FAI-11031, FAI-11032: Add Jira project versions and version issues streams

### PR DESCRIPTION
**Pull request by chalenge**
## Description

- Add Jira project versions and version issues streams

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/jira-source/test_files/project_version_issues.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "fields": {
+    "created": "2024-03-14T12:00:00Z",
+    "updated": "2024-03-14T12:00:00Z"
+    }
+  },
+  {
+    "id": "2",
+    "key": "TEST-2",
+    "fields": {
+    "created": "2024-03-14T12:00:00Z",
```
Comment (matiaslcoulougian) [inline sources/jira-source/test_files/project_version_issues.json:14]: Nit: linting

File: sources/jira-source/src/streams/faros_project_versions.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    for (const version of await jira.getProjectVersions(projectKey)) {
+      yield {
+        ...pick(version, [
+          'id',
+          'description',
+          'name',
+          'startDate',
+          'releaseDate',
+          'projectId',
```
Comment (matiaslcoulougian) [inline sources/jira-source/src/streams/faros_project_versions.ts:35]: Are we using projectId property?

File: sources/jira-source/src/jira.ts [hunk 3]
```diff
@@ -823,6 +822,13 @@ export class Jira {
     }
   }
 
+  async getBoardsFromCloud(projectId: string): Promise<void> {
```
Comment (matiaslcoulougian) [inline sources/jira-source/src/jira.ts:825]: Remove this unused function

**Review by matiaslcoulougian — CHANGES_REQUESTED** [review]

File: sources/jira-source/src/streams/faros_project_versions.ts (hunk 2 above)
Comment (chalenge) [reply sources/jira-source/src/streams/faros_project_versions.ts:35]: It is a required property on ProjectVersion via Version2Models.Version

**Pull request merged**